libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -179,6 +183,49 @@ bool IRGeneratorForStatements::visit(FunctionCall const& _functionCall)
 	return false;
 }
 
+bool IRGeneratorForStatements::visit(InlineAssembly const& _inlineAsm)
+{
+	struct CopyTranslate: public yul::ASTCopier
+	{
+		using ExternalRefsMap = std::map<yul::Identifier const*, InlineAssemblyAnnotation::ExternalIdentifierInfo>;
+
+		CopyTranslate(IRGenerationContext& _context, ExternalRefsMap _references):
+			m_context(_context), m_references(_references) {}
+
+		yul::YulString translateIdentifier(yul::YulString _name) override
+		{
+			for (auto const& pair: m_references)
```

Comment:
You could avoid running this loop for every single identifier by overriding `Expression operator()(yul::Identifier const&)`

Reply:
Wasn't as straight forward as I hoped, but should be good now